workflow-templates/copyright-update.yml

```diff
@@ -0,0 +1,48 @@
+#
+# Copyright 2017-2021 the original author or authors from the JHipster project.
+#
+# This file is part of the JHipster project, see https://www.jhipster.tech/
+# for more information.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Copyright Update
+on:
+  schedule:
+    - cron: '0 0 31 12 *' # Repeats December 31st every year
+jobs:
+  pipeline:
+    name: copyright update
+    runs-on: ubuntu-latest
+    timeout-minutes: 40
+
+    steps:
+      # Checkout
+      - uses: actions/checkout@v2
+
+      # Update the copyright headers
+      - name: Find and Replace
+        run: |
+          CURRENT_YEAR=$(date +'%Y')
+          NEW_YEAR=$(($CURRENT_YEAR + 1))
+          grep -rlZ "Copyright 2017-$CURRENT_YEAR" . | xargs -0 sed -i "s/Copyright 2017-$CURRENT_YEAR/Copyright 2017-$NEW_YEAR/g"
```

Comment:
are you sure it can work with 2017 ?
some projects are older

Reply:
@pascalgrimaud : Thanks for the review. Good catch. 👍🏽 I just copied and pasted from `jhipster-online`. Just corrected this to take an arbitrary year. Now it should be good. 😄